e2e/utils/test-utils.ts

```diff
@@ -267,6 +255,22 @@ export async function findOrCreateWorkspaceCard(
   return cardFound;
 }
 
+/**
+ * Find a suitable workspace among existing workspaces with OWNER role and older than 30 minutes.
+ */
+export async function findAllCards(page: Page): Promise<WorkspaceCard[]> {
+  const existingCards: WorkspaceCard[] = await WorkspaceCard.findAllCards(page, WorkspaceAccessLevel.Owner);
+  // Filter to include Workspaces older than 30 minutes
+  const halfHourMilliSec = 1000 * 60 * 30;
+  const now = Date.now();
+  return Promise.all(
+    await filter(existingCards, async (card) => {
+      const workspaceTime = Date.parse(await card.getLastChangedTime());
+      return now - workspaceTime > halfHourMilliSec;
+    })
+  );
```

Comment:
I don't think this is any clearer than what you have, but it prevents the use of reduce, which is a powerful abstraction.
I am just putting it in because I needed to write it out to see what it would look like. Only adding it as a reference.

```javascript
  const halfHourAgoMillis = Date.now() - 1000 * 60 * 30;
  return fp.flow(
    fp.map<WorkspaceCard, Promise<[number, WorkspaceCard]>>(async (card) => [Date.parse(await card.getLastChangedTime()), card]),
    promises => Promise.all(promises),
    fp.filter(([lastChangedMillis]) => halfHourAgoMillis > lastChangedMillis),
    fp.map<[number, WorkspaceCard], WorkspaceCard>(([, card]) => card)
  )(existingCards)
```

I do like your filter function though, because it can be generally usefull